src/registry.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Rule, RuleCollection} from './rule';
+import {stripWhitespace} from './util';
+
+/**
+ * A singleton class where lint rules and rule collections can register
+ * themselves and you can get a collection of rules by querying.
+ */
+export class LintRegistry {
+  private static readonly _singleton = new LintRegistry();
```

Comment:
I don't think you really need to both expose a registry variable and try to make the class a singleton in it's implementation. Exporting the `registry` binding is enough. It's similar to `CustomElementRegistry`, there's a built-in global singleton, but you could create other instances if you wanted.